core/src/main/java/com/netflix/conductor/core/execution/DeciderService.java

```diff
@@ -260,21 +256,31 @@ private DeciderOutcome decide(final WorkflowDef workflowDef, final Workflow work
         return Collections.singletonList(rerunFromTask);
     }
 
-    private void updateOutput(final WorkflowDef def, final Workflow workflow) {
-
+    void updateOutput(final WorkflowDef def, final Workflow workflow, @Nullable Task task) {
         List<Task> allTasks = workflow.getTasks();
```

Comment:
Why get all tasks when task is not null?

Reply:
This is so that the workflow output is only updated only when the workflow has tasks in its execution.